packages/dds/tree/src/feature-libraries/modular-schema/genericFieldKind.ts

```diff
@@ -202,13 +203,20 @@ export function convertGenericChange<TChange>(
 		makeAnonChange(target.editor.buildChildChange(index, nodeChange)),
 	);
 
-	return target.rebaser.compose(
-		perIndex,
-		composeChild,
-		genId,
-		invalidCrossFieldManager,
-		revisionMetadata,
-	);
+	return perIndex.reduce(
+		(a, b) =>
+			makeAnonChange(
+				target.rebaser.compose(
+					a,
+					b,
+					composeChild,
+					genId,
+					invalidCrossFieldManager,
+					revisionMetadata,
+				),
+			),
+		makeAnonChange(target.createEmpty()),
```

Comment:
Instead of starting with an empty change we should return early if the list is empty.